### ISSUE TITLE
change the way how providers are initialized - create a provider constructor

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2043]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2039
- Created at 2013-07-30 11:27:32 by [pbrezina](https://pagure.io/user/pbrezina)
- Closed at 2020-03-11 15:36:58 as Invalid
- Assigned to nobody
---

Each provider consists of several modules (id, auth, chpass, autofs, ...). These modules are initialized by sssm_(provider)_(module)_init() function, however all modules calls init function of id module to initialize shared resources.

We should create a common provider constructor (say sssm_(provider)_init) that is invoked only once by backend, when a first module of this constructor is beeing loaded.

### Comments

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2013-08-01 15:25:19](https://pagure.io/SSSD/sssd/issue/2039#222815)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.13 beta
rhbz:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2014-09-26 18:49:58](https://pagure.io/SSSD/sssd/issue/2039#222816)**

Fields changed

mark:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2015-02-10 16:43:32](https://pagure.io/SSSD/sssd/issue/2039#222817)**

Fields changed

milestone: SSSD 1.13 beta => SSSD 1.13 backlog
priority: major => minor

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2015-02-12 20:27:08](https://pagure.io/SSSD/sssd/issue/2039#222818)**

Mass-moving tickets not planned for the next two releases.

Please reply with a comment if you disagree about the move..

milestone: SSSD 1.13 backlog => SSSD 1.15 beta

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-02-24 14:26:50](https://pagure.io/SSSD/sssd/issue/2039#222819)**

**Metadata Update from @pbrezina**:
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2020-03-11 15:36:59](https://pagure.io/SSSD/sssd/issue/2039#632517)**

**Metadata Update from @pbrezina**:
- Custom field design_review adjusted to on (was: 0)
- Custom field mark adjusted to on (was: 0)
- Custom field patch adjusted to on (was: 0)
- Custom field review adjusted to on (was: 0)
- Custom field sensitive adjusted to on
- Custom field testsupdated adjusted to on (was: 0)
- Issue close_status updated to: Invalid
- Issue status updated to: Closed (was: Open)

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2020-03-11 15:37:16](https://pagure.io/SSSD/sssd/issue/2039#632518)**

Already done.
